src/DataTypes/Student.jl

```diff
@@ -0,0 +1,40 @@
+mutable struct Student
+    id::Int                     # Unique ID for student
+    total_credits::Int          # The total number of credit hours the student has earned
+    gpa::Float64                # The student's GPA
+    total_points::Float64       # The total number of points the student has earned
+
+    attributes::Dict            # A dictionary that can store any kind of student attribute
+    stopout::Bool               # Indicates whether the student has stopped out. (False if the student has, True if still enrolled)
+    stopsem::Bool               # The term the student stopped out.
+    termcredits::Int            # The number of credits the student has enrolled in for a given term.
+    performance::Dict           # Stores the grades the student has made in each course.
+    graduated::Bool             # Indicates wheter the student has graduated.
+    gradsem::Int                # The term the student has graduated.
+    termpassed::Array{Int}      # An array that represents the term in which the student passed each course.
```

Comment:
I think that we should adjust this implementation to use the StudentRecord type since many of the fields are similar or even the same. We may need to work through some adjustments / additions to the StudentRecord, and something like `graduated` may stay in this struct. From an OOP standpoint "Student s has a StudentRecord sr"

Reply:
We could add a metadata field to StudentRecord to capture those temporary variables for simulation

Reply:
That's a great idea. I think some of the fields should actually be incorporated into the StudentRecord, but others may belong in a metadata dictionary. Let's discuss this with Greg on a call sometime to see what he thinks?